demo/admin/src/products/future/AssignProductsGrid.tsx

```diff
@@ -0,0 +1,82 @@
+import { gql, useApolloClient, useQuery } from "@apollo/client";
+import { Savable } from "@comet/admin";
+import { CircularProgress } from "@mui/material";
+import {
+    GQLGetProductIdsForProductCategoryQuery,
+    GQLGetProductIdsForProductCategoryQueryVariables,
+    GQLSetProductCategoryMutation,
+    GQLSetProductCategoryMutationVariables,
+} from "@src/products/future/AssignProductsGrid.generated";
+import { ProductsGrid as SelectProductsGrid } from "@src/products/future/generated/SelectProductsGrid";
+import isEqual from "lodash.isequal";
+import React, { useEffect, useMemo, useState } from "react";
+import { FormattedMessage } from "react-intl";
+
+const setProductCategoryMutation = gql`
+    mutation SetProductCategory($id: ID!, $input: [ID!]!) {
+        updateProductCategory(id: $id, input: { products: $input }) {
+            id
+        }
+    }
+`;
+
+const getProductIdsForProductCategory = gql`
+    query GetProductIdsForProductCategory($id: ID!) {
+        products(filter: { category: { equal: $id } }) {
+            nodes {
+                id
+            }
+        }
+    }
+`;
+
+interface FormProps {
+    productCategoryId: string;
+}
+
+export function AssignProductsGrid({ productCategoryId }: FormProps): React.ReactElement {
+    const client = useApolloClient();
+
+    const { data, error, loading } = useQuery<GQLGetProductIdsForProductCategoryQuery, GQLGetProductIdsForProductCategoryQueryVariables>(
+        getProductIdsForProductCategory,
+        {
+            variables: { id: productCategoryId },
+        },
+    );
+
+    const initialValues = useMemo(() => {
+        return data?.products.nodes ? data.products.nodes.map((product) => product.id) : [];
+    }, [data]);
+    const [values, setValues] = useState<string[]>(initialValues);
+    useEffect(() => {
+        setValues(initialValues);
+    }, [initialValues]);
```

Comment:
As mentioned [here](https://github.com/vivid-planet/comet/pull/2428#discussion_r1730928421), this is probably too complicated...